src/components/extensive/Tables/DisturbancesTable.tsx

```diff
@@ -35,12 +41,42 @@ const DisturbancesTable = ({ modelName, modelUUID, collection, onFetch }: Distur
   );
 
   const hasAllHeaders = !!disturbances?.data?.[0]?.extent && !!disturbances?.data?.[0].intensity;
+  const { data: record } = useGetV2FormsENTITYUUID<{ data: GetV2FormsENTITYUUIDResponse }>({
+    pathParams: {
+      uuid: modelUUID,
+      entity: modelName
+    }
+  });
+
+  const viewDataDisturbances = record?.data?.form?.form_sections.map((questions: any) =>
+    questions.form_questions.map((item: any) => item.uuid).map((item: any) => record?.data?.answers?.[item])
+  );
+
+  const verifydata = () => {
+    record?.data?.form?.form_sections.forEach((section: any, sectionIndex: number) => {
+      section.form_questions.forEach((question: any, questionIndex: number) => {
+        if (question.children) {
+          question.children.forEach((child: any) => {
+            if (child.input_type === "disturbances") {
+              setShowDisturbance(viewDataDisturbances?.[sectionIndex as number]?.[questionIndex as number]);
+              viewDataDisturbances?.[sectionIndex as number]?.[questionIndex as number];
+            }
+          });
+        }
+      });
+    });
+    return false;
+  };
 
+  useEffect(() => {
+    verifydata();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [record]);
```

Comment:
I would say that this useEffect is not needed, you can replace this with:

```
const showDisturbance = useMemo(() => {
    //all the code inside verifydata 
    for (let section in record?.data?.form?.form_sections){
        ....
    }
    return false;
}, [record])
```

by the way, `verifydata` is not a good function name anyways.